### PR TITLE
MINOR: Further code cleanups.

### PR DESCRIPTION
**Pull request by dguy**
Remove redundant code, make fields private, remove non thrown exceptions etc

### CODE REVIEW

**Comment by dguy** [comment]
@hjafarpour

File: ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analysis.java [hunk 1]
```diff
@@ -46,7 +46,7 @@
 
   private Expression havingExpression = null;
 
-  private Optional<Integer> limitClause = Optional.empty();
```
Comment (satybald) [inline ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analysis.java]: why would you leave Optional? Why not use functional style and avoid dealing with NPE?

File: ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java [hunk 2]
```diff
@@ -234,10 +235,14 @@ public KStream getKstream() {
 
   public String getExecutionPlan(String indent) {
     StringBuilder stringBuilder = new StringBuilder();
-    stringBuilder.append(indent + " > [ " + type + " ] Schema: " + SchemaUtil
-        .getSchemaDefinitionString(schema) + ".\n");
+    stringBuilder.append(indent)
+        .append(" > [ ")
+        .append(type).append(" ] Schema: ")
+        .append(SchemaUtil
+        .getSchemaDefinitionString(schema)).append(".\n");
```
Comment (satybald) [inline ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java]: wouldn't it be better to make method invocation `SchemaUtil.getSchemaDefinitionString(schema)` and object definition on the same line?

File: ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analysis.java (hunk 1 above)
Comment (dguy) [reply ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analysis.java]: It won't be null as users of this class will call `getLimitClause()` that returns `Optional.ofNullable(limitClause)` - there is no reason for it to be `Optional` within the class

File: ksql-serde/src/main/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroSerializer.java [hunk 3]
```diff
@@ -37,32 +36,19 @@
...
 
   public KsqlGenericRowAvroSerializer(org.apache.kafka.connect.data.Schema schema,
                                       SchemaRegistryClient schemaRegistryClient, KsqlConfig
-                                          ksqlConfig, boolean isInternal) {
-    if (isInternal) {
-      this.schema = SchemaUtil.getAvroSerdeKsqlSchema(schema);
-    } else {
-      this.schema = SchemaUtil.getSchemaWithNoAlias(schema);
-    }
-
```
[the remark this answers is not in the document]
Comment (dguy) [reply ksql-serde/src/main/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroSerializer.java]: It is removed as `this.schema` is never used so there is no point in doing the `if(...)`

File: ksql-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java [hunk 4]
```diff
@@ -122,13 +122,11 @@ private AggregateNode buildAggregateNode(final Schema inputSchema,
     return new AggregateNode(new PlanNodeId("Aggregate"),
                              sourcePlanNode,
                              aggregateSchema,
-                             analysis.getSelectExpressions(),
-                             analysis.getGroupByExpressions(),
+        analysis.getGroupByExpressions(),
```
Comment (rodesai) [inline ksql-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java]: nit: spacing

**Comment by dguy** [comment]
retest this please

**Comment by dguy** [comment]
retest this please

**Comment by dguy** [comment]
retest this please

**Comment by dguy** [comment]
retest this please

**Pull request merged**